airflow-core/src/airflow/configuration.py

```diff
@@ -1132,7 +1179,8 @@ def _get_option_from_commands(
         section: str,
         issue_warning: bool = True,
         extra_stacklevel: int = 0,
-    ) -> str | None:
+        **kwargs,
+    ) -> str | object:
```

Comment:
I don't love this being typed as `| object` -- perhaps we should do


```python
class ValueNotFound: pass
VALUE_NOT_FOUND_SENTINEL = ValueNotFound()
```

Then this could be 

```suggestion
    ) -> str | ValueNotFound:
```

Not a big thing, but might be worth it

Reply:
Let me make that change, I agree with you.

Reply:
Handled in [comments from ash](https://github.com/apache/airflow/pull/57970/commits/4b94ccaae63dd564c05151c7967e18ee5966d5cc)